BinaryEncoding.md

```diff
@@ -149,18 +176,44 @@ ID: `import`
 The import section declares all imports that will be used in the module.
 
 | Field | Type | Description |
-| ----- |  ----- | ----- |
+| ----- | ---- | ----------- |
 | count | `varuint32` | count of import entries to follow |
 | entries | `import_entry*` | repeated import entries as described below |
 
 #### Import entry
 | Field | Type | Description |
-| ----- |  ----- | ----- |
-| sig_index | `varuint32` | signature index of the import |
+| ----- | ---- | ----------- |
 | module_len | `varuint32` | module string length |
 | module_str | `bytes` | module string of `module_len` bytes |
-| function_len | `varuint32` | function string length |
-| function_str | `bytes` | function string of `function_len` bytes |
+| field_len | `varuint32` | field name length |
+| field_str | `bytes` | field name string of `field_len` bytes |
+| kind | `definition_kind` | the kind of definition being imported |
+
+Followed by, if the `kind` is `Function`:
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| sig_index | `varuint32` | signature index of the import |
+
+or, if the `kind` is `Table`:
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| element_type | `varuint7` | `0x40`, indicating [`anyfunc`](AstSemantics.md#table) |
```

Comment:
Agreed. A while back I spent some thought on what the Wasm type structure would become with GC types (happy to talk about that next week). It quickly explodes into a zoo of categories like num_type, ref_type, value_type, field_type, mem_type, struct_type etc that combine in various ways. So I would rather avoid being too cute now, especially since we have plenty of room in the type opcode space. Perhaps just pick some free opcode with enough distance from others.